libbindgen/src/lib.rs

```diff
@@ -509,6 +509,25 @@ impl<'ctx> Bindings<'ctx> {
                     -> Result<Bindings<'ctx>, ()> {
         let span = span.unwrap_or(DUMMY_SP);
 
+        // TODO: Make this path fixup configurable?
+        if let Some(clang) = clang_sys::support::Clang::find(None) {
+            // If --target is specified, assume caller knows what they're doing
+            // and don't mess with include paths for them
+            let has_target_arg = options.clang_args.iter()
+                .rposition(|arg| arg.starts_with("--target"))
+                .is_some();
+            if !has_target_arg {
+                // TODO: distinguish C and C++ paths? C++'s should be enough, I
+                // guess.
+                for path in clang.cpp_search_paths.into_iter() {
```

Comment:
I start wondering (again) why do we need this fixup. What would happen if we don't have it? It doesn't seem to me it would fail to build without this on macOS (since I can run stylo bindgen with `--target=x86_64-apple-darwin -stdlib=libc++` specified), and we've already disabled it for stylo build on Windows (because we always specify `--target`). So Linux?

My guess is that the user of this library should always specify things they need, e.g. `-stdlib=libc++` or `-stdlib=libstdc++`, rather than we doing hacky fixup under the hood.

Also, other than `--target`, there are many clang arguments can affect search path selection as well, e.g. `-stdlib`, and `-mmacos-version-min` on macOS, which means the paths here may not match what user expect, and thus add noise.

The recent example of bustage with `MACOSX_DEPOLYMENT_TARGET` environment is probably another example of harmfulness this kind of implicit fixup adds. If we have `-stdlib=libc++` specified since the very beginning because of the lack of this fixup, that may never happen.